scripts/json2db.pl

```diff
@@ -0,0 +1,238 @@
+
+my $motionadd = $dbh->prepare("INSERT INTO policydivisions (division_id, policy_id, house, direction, division_title, yes_text, no_text, division_date, division_number, gid) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)");
+my $motionupdate = $dbh->prepare("UPDATE policydivisions SET gid = ?, division_title = ?, yes_text = ?, no_text = ?, direction = ? WHERE division_id = ? AND policy_id = ?");
+
+my $votecheck = $dbh->prepare("SELECT member_id, vote FROM memberdivisionvotes WHERE division_id = ?");
+my $voteadd = $dbh->prepare("INSERT INTO memberdivisionvotes (member_id, division_id, vote) VALUES (?, ?, ?)");
+my $voteupdate= $dbh->prepare("UPDATE memberdivisionvotes SET vote = ? WHERE member_id = ? AND division_id = ?");
+
+my $motionsdir = $parldata . "scrapedjson/policy-motions/";
+
+$motion_count = $policy_count = $vote_count = 0;
+
+foreach my $dreamid (
+    363,
+    810,
+    811,
+    826,
+    837,
+    975,
+    984,
+    996,
+    1027,
+    1030,
+    1049,
+    1050,
+    1051,
+    1052,
+    1053,
+    1065,
+    1071,
+    1074,
+    1079,
+    1084,
+    1087,
+    1105,
+    1109,
+    1110,
+    1113,
+    1120,
+    1124,
+    1132,
+    1136,
+    6667,
+    6670,
+    6671,
+    6672,
+    6673,
+    6674,
+    6676,
+    6677,
+    6678,
+    6679,
+    6680,
+    6681,
+    6682,
+    6683,
+    6684,
+    6685,
+    6686,
+    6687,
+    6688,
+    6690,
+    6691,
+    6692,
+    6693,
+    6694,
+    6695,
+    6696,
+    6697,
+    6698,
+    6699,
+    6702,
+    6703,
+    6704,
+    6705,
+    6706,
+    6707,
+    6708,
+    6709,
+    6710,
+    6711,
+    6715,
+    6716,
+    6718,
+    6719,
+    6720,
+    6721
+) {
+    my $policy_file = $motionsdir . $dreamid . ".json";
+    if ( ! -f $policy_file ) {
+        warn "no json file for policy $dreamid at $policy_file";
+        next;
+    }
+    my $policy_json = read_file($policy_file);
+    my $policy = $json->decode($policy_json);
+    my $motions = @{ $policy->{aspects} };
+
+
+    my $curr_policy = $dbh->selectrow_hashref($policycheck, {}, $dreamid);
+    if ( $curr_policy ) {
+        $policyupdate->execute($policy->{title} || '', $policy->{text}, $dreamid);
+    } else {
+        $policyadd->execute($dreamid, $policy->{title} || '', $policy->{text});
+    }
+
+    $policy_count++;
+
+    for my $motion ( @{ $policy->{aspects} } ) {
+        $motion_count++;
+        my ($motion_num, $house);
+        ($motion_num = $motion->{motion}->{id}) =~ s/pw-\d+-\d+-\d+-(\d+)/$1/;
+        ($house = $motion->{motion}->{organization_id}) =~ s/uk.parliament.(\w+)/$1/;
+
+        my $sources = $motion->{motion}->{sources};
+        my $gid = '';
+        foreach my $source (@$sources) {
+            if ( defined $source->{gid} ) {
+                $gid = $source->{gid};
+            }
+        }
+
+        my $motion_id = $motion->{motion}->{id} . "-$house";
```

Comment:
I don't think this is unique (not even by policy perhaps), as the ID repeats every day in the Lords, and every Parliament in the Commons. If the house is included in the ID (as I think it has to be, as otherwise it may not be unique), then you might want to use the ID as is?